application/core/LSYii_Application.php

```diff
@@ -508,6 +511,57 @@ public function getTwigCustomExtensionsConfig($sUsertwigextensionrootdir, $aAppl
         return $aApplicationConfig;
     }
 
+    /**
+     * @inheritdoc
+     * Special handling for SEO friendly URLs
+        */
+    public function createController($route, $owner=null)
+    {
+        $controller = parent::createController($route, $owner);
+
+        // If no controller is found by standard ways, check if the route matches
+        // an existing survey's alias.
+        if (is_null($controller)) {
+            $controller = $this->createControllerFromShortUrl($route);
+        }
+
+        return $controller;
+    }
+
+    /**
+     * Create controller from short url if the route matches a survey alias.
+     * @param string $route the route of the request.
+     * @return array<mixed>|null
+     */
+    private function createControllerFromShortUrl($route)
```

Comment:
Do we have a unit-test for this function?

Reply:
Not a unit test, but a functional test that uses it.
We can do a Unit test if needed